apps/src/gamelab/ResizablePanes.jsx

```diff
@@ -0,0 +1,139 @@
+/** @file Divide a region into a set of panels and allow resizing them.
+ *
+ *  Similar to react-split-pane (MIT license) but supports more than two panes
+ *  and works with React 0.14.7.
+ *  @see https://github.com/tomkp/react-split-pane
+ */
+'use strict';
+
+var _ = require('../lodash');
+
+/**
+ * Wraps its children to display them in a flexbox layout.
+ */
+var ResizablePanes = React.createClass({
+  propTypes: {
+    style: React.PropTypes.object
+  },
+
+  getInitialState: function () {
+    return {
+      dragging: false,
+      index: 0,
+      overrideSizes: {}
+    };
+  },
+
+  componentDidMount() {
+    document.addEventListener('mousemove', this.onMouseMove);
+    document.addEventListener('mouseup', this.onMouseUp);
+  },
+
+  componentWillUnmount() {
+    document.removeEventListener('mousemove', this.onMouseMove);
+    document.removeEventListener('mouseup', this.onMouseUp);
+  },
+
+  onResizerMouseDown: function (event) {
+    this.unFocus();
+    this.setState({
+      dragging: true,
+      index: parseInt(event.target.dataset.resizerIndex, 10)
+    });
+  },
+
+  onMouseMove(event) {
+    if (!this.state.dragging) {
+      return;
+    }
+    this.unFocus();
+    var resizingPane = this.refs['pane-' + this.state.index];
+    if (!resizingPane) {
+      return;
+    }
+
+    var resizingPaneDOMNode = ReactDOM.findDOMNode(resizingPane);
+    if (!resizingPaneDOMNode.getBoundingClientRect) {
+      return;
+    }
+
+    var boundingRect = resizingPaneDOMNode.getBoundingClientRect();
+    var newWidth = event.clientX - boundingRect.left;
+    var overrideSizesChange = {};
+    overrideSizesChange[this.state.index] = newWidth;
+
+    this.setState({
+      overrideSizes: _.assign({}, this.state.overrideSizes, overrideSizesChange)
+    });
+  },
+
+  onMouseUp() {
+    if (this.state.dragging) {
+      this.setState({
+        dragging: false
+      });
```

Comment:
For single field changes like this, I'll usually just put it on one line. Thoughts?
```
this.setState({ dragging: false });